benchconnect/README.md

```diff
@@ -0,0 +1,42 @@
+# benchconnect
+
+A small Python package that creates a CLI for benchmark runners interacting
+with a Conbench server.
+
+This is a light wrapper on top of
+[benchadapt.BenchmarkResult](https://github.com/conbench/conbench/blob/main/benchadapt/python/benchadapt/result.py),
+[benchadapt.BenchmarkRun](https://github.com/conbench/conbench/blob/main/benchadapt/python/benchadapt/run.py), and
+[benchclients.ConbenchClient](https://github.com/conbench/conbench/blob/main/benchclients/python/benchclients/conbench.py).
+If your code is already in Python, it may be simpler to use those classes
+directly.
+
+## Installation
+
+With [pipx](https://pypa.github.io/pipx/):
+
+```bash
+pipx install benchconnect@git+https://github.com/conbench/conbench.git@main#subdirectory=benchconnect
+```
+
+## Usage
+
+See the man pages:
+
+```shell
+benchconnect --help
+benchconnect augment --help
+benchconnect post --help
+benchconnect put --help
+```
+
+The `augment` command takes JSON for a benchmark result (`augment result`)
+or benchmark run (`augment run`) and fills in defaults for unspecified fields.
+This is useful for filling in fields like `machine_info` consistently. Note
+this is stateless, and will not make `run_id` and `suite_id` values correspond
+across results, nor will the resulting JSON necessarily be complete enough to
+post to a Conbench server correctly (it can't fill in things like `run_reason`
+for you); please ensure data is correct before posting.
+
+The `post` and `put` commands authenticate with a Conbench server (see man
+pages for environment variables) and send JSON passed. Since they correspond
+to the API, there are `post result` and `post run` methods, but only `put run`.
```

Comment:
It would be nice to have an example in here. I started working on one, and in doing so, wondering if it would make sense to have a slightly higher-level API (basically, abstract the REST bits, giving folks an entry point that are the more-or-less logical operations they are doing)

```
benchconnect run start --run-reason="my fancy run"

# you can send either a path to results
benchconnect result send --path result.json
# or the json itself
benchconnect result send --json="{...}"

benchconnect run finish
```

And further, if you want to see how results will be augmented with default data, you can:

```
benchconnect run start --run-reason="my fancy run" --dry-run
benchconnect result augment --path result.json
```

I've made up a few things in here that would be necessary to get some of this working (e.g. having a `dry-run` option on starting a run that doesn't send to the server, but does setup the state that is needed to persist things like that run reason). I'm also presuming that `benchconnect result send` will automatically augment. The names of these.

On storing state: I know you've brought up multiple times in this PR that this is (currently) stateless. IMO that's making things harder on ourselves than it needs to be (we have to worry about passing things around, or making sure we always get run-reason in the right places). What if we did the simplest possible thing and when we start a run, we save a statefile (working directory is totally fine) that includes these bits? Then we don't have to worry about passing them through the stack.

Reply:
I do like this api, and it would solve the ndjson problem

Reply:
K, added a new stateful workflow, please take a look when you get a minute